### PR TITLE
Fix countItems in DataFilterExtension

### PR DESCRIPTION
**Pull request by felixpalmer**
#### Background

`countItems` has not been working since the upgrade to v9

#### Change List
- Remove deprecated luma APIs
- Set buffer layout on `filterModel` creation
- Update test to catch this regression in the future

### CODE REVIEW

File: modules/extensions/src/data-filter/aggregator.ts [hunk 1]
```diff
@@ -19,7 +18,8 @@ const float component = 1.0 / 255.0;
 
 void main() {
   #ifdef FLOAT_TARGET
-    dataFilter_value *= float(filterIndices != filterPrevIndices);
+    // BUG: always 0 for some reason
+    // dataFilter_value *= float(filterIndices != filterPrevIndices);
```
Comment (felixpalmer) [inline modules/extensions/src/data-filter/aggregator.ts:22]: @Pessimistress any idea why this is broken? I've tried different variants of the check and it always evaluates to 0. Perhaps there is something wrong with using `vertexOffset`?

**Review by ibgreen — APPROVED** [review]
Thanks for this, LGTM.

File: modules/extensions/src/data-filter/aggregator.ts (hunk 1 above)
Comment (Pessimistress) [reply modules/extensions/src/data-filter/aggregator.ts:22]: Subtle catch of v9 - when an attribute's descriptor contains `shaderAttributes`, the sub-attributes cannot use the same name.

**Review by Pessimistress — APPROVED** [review]

File: modules/extensions/src/data-filter/aggregator.ts (hunk 1 above)
Comment (felixpalmer) [reply modules/extensions/src/data-filter/aggregator.ts]: That is subtle indeed :) would be nice if the types could catch this

**Pull request merged**